src/psyclone/tests/test_files/dynamo0p3/testkern_access_write_mod.f90

```diff
@@ -0,0 +1,106 @@
+! modification, are permitted provided that the following conditions are met:
+!C
+! * Redistributions of source code must retain the above copyright notice, this
+!   list of conditions and the following disclaimer.
+!
+! * Redistributions in binary form must reproduce the above copyright notice,
+!   this list of conditions and the following disclaimer in the documentation
+!   and/or other materials provided with the distribution.
+!
+! * Neither the name of the copyright holder nor the names of its
+!   contributors may be used to endorse or promote products derived from
+!   this software without specific prior written permission.
+!
+! THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+! AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+! IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+! DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+! FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+! DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+! SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+! CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+! OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+! OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+! -----------------------------------------------------------------------------
+! Modified: J. Henrichs, Bureau of Meteorology
+
+!> @brief Kernel which assembles a locally assembled matrix (LMA) into a
+!!        columnwise assembled matrix (CMA).
+module testkern_access_write_mod
+
+use kernel_mod,              only : kernel_type
+use argument_mod,            only : arg_type, func_type,                 &
+                                    GH_OPERATOR, GH_COLUMNWISE_OPERATOR, &
+                                    GH_REAL, GH_READ, GH_WRITE,          &
+                                    ANY_DISCONTINUOUS_SPACE_1,           &
+                                    ANY_DISCONTINUOUS_SPACE_2,           &
+                                    CELL_COLUMN
+
+use constants_mod,           only : r_def, r_solver, i_def
+
+implicit none
+
+private
+
+!-------------------------------------------------------------------------------
+! Public types
+!-------------------------------------------------------------------------------
+
+type, public, extends(kernel_type) :: testkern_access_write_type
+  private
+  type(arg_type) :: meta_args(3) = (/                                  &
+       arg_type(GH_FIELD,               GH_REAL, GH_READ,              &
+                ANY_DISCONTINUOUS_SPACE_1),                            &
+       arg_type(GH_OPERATOR,            GH_REAL, GH_READ,              &
+                ANY_DISCONTINUOUS_SPACE_1, ANY_DISCONTINUOUS_SPACE_2), &       
+       arg_type(GH_COLUMNWISE_OPERATOR, GH_REAL, GH_WRITE,             &
```

Comment:
If it's all about writing to an operator, I'd have thought we'd already have an example of this.

Reply:
This file is removed now.